src/libraries/Microsoft.Extensions.DependencyInjection/src/CallSiteJsonFormatter.cs

```diff
@@ -130,26 +130,26 @@ public CallSiteFormatterContext(StringBuilder builder, int offset, HashSet<Servi
             public int Offset { get; }
             public StringBuilder Builder { get; }
 
-            public bool ShouldFormat(ServiceCallSite serviceCallSite)
+            public readonly bool ShouldFormat(ServiceCallSite serviceCallSite)
```

Comment:
This is an example of a kind of change that makes me a tad nervous (even if ultimately we decide this case is ok).  This method does mutate, it's just mutating state in a stored reference type.  If the implementation were to change in the future, such that some of this state were actually stored as fields in the struct, it would no longer be readonly.

Reply:
Yeah, that's true, but...mostly I can't decide by myself is it ok or not :( . So it seems like I have to mark those methods anyway and wait here for review, isn't it?
But for the start I can skip all methods that contains .Add/.Append and so on to minimize false negative cases

Reply:
> So it seems like I have to mark those methods anyway and wait here for review, isn't it?

Yup, a reasonable approach as you suggest is to programmatically mark for review all that could potentially be made readonly, and then back off from that list based on fully reviewing it prior to committing.